table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+     * @param pattern pattern
+     * @return this builder
+     */
+    public Builder addBypassedTable(Pattern pattern) {
+      return addTable(new PatternNameTemplate(pattern), Entry.BYPASS);
+    }
+
+    /**
+     * Add a table by its exact name to the ignore entry.
+     * @param name table name
+     * @return this builder
+     */
+    public Builder addIgnoredTable(String name) {
+      return addTable(new ExactNameTemplate(name), Entry.IGNORE);
+    }
+
+    /**
+     * Add a table by pattern to the ignore entry.
+     * @param pattern pattern
+     * @return this builder
+     */
+    public Builder addIgnoredTable(Pattern pattern) {
+      return addTable(new PatternNameTemplate(pattern), Entry.IGNORE);
+    }
+
+    private Builder addTable(NameTemplate item, Entry entry) {
+      switch (entry) {
+        case BYPASS:
+          mBypassedTables.add(item);
+          break;
+        case IGNORE:
+          mIgnoredTables.add(item);
+          break;
+        case NONE:
+        default:
+          // unreachable
+      }
+      return this;
+    }
+
+    /**
+     * Include or exclude the partitions of a bypassed tabls.
+     * @param tableName name of the parent table
+     * @param mode {@link Mode} of the tables
+     * @return this builder
+     */
+    public Builder setBypassedPartitionsMode(String tableName, Mode mode) {
+      mPartitionBypassModes.put(tableName, mode);
+      return this;
+    }
+
+    /**
+     * Add a partition by its exact name.
+     * @param tableName the containing table
+     * @param partitionName partition name
+     * @return this builder
+     */
+    public Builder addBypassedPartition(String tableName, String partitionName) {
+      return addBypassedPartition(tableName, new ExactNameTemplate(partitionName));
+    }
+
+    /**
+     * Add a partition by pattern.
+     * @param tableName the containing table
+     * @param partitionPattern partition pattern
+     * @return this builder
+     */
+    public Builder addBypassedPartition(String tableName, Pattern partitionPattern) {
+      return addBypassedPartition(tableName, new PatternNameTemplate(partitionPattern));
+    }
+
+    private Builder addBypassedPartition(String tableName, NameTemplate partition) {
+      mBypassedPartitions.compute(tableName, (tblName, set) -> {
+        if (set == null) {
+          return Sets.newHashSet(partition);
+        } else {
+          set.add(partition);
+          return set;
+        }
+      });
+      return this;
+    }
+
+    /**
+     * Build a {@link UdbFilterSpec}.
+     * The builder should not be used after this method returns.
+     * @return {@link UdbFilterSpec}
+     * @throws IllegalArgumentException when invalid or conflicting configurations
+     *                                          have been provided
+     */
+    public UdbFilterSpec build() {
+      ImmutableMap.Builder<String, Pair<Mode, Set<NameTemplate>>> partitionsBuilder
+          = ImmutableMap.builder();
```

Comment:
```suggestion
      ImmutableMap.Builder<String, Pair<Mode, Set<NameTemplate>>> partitionsBuilder =
            ImmutableMap.builder();
```

Reply:
Fixed in 391b8b369413e0ba5c9528c2c0645cc54ee76932.